hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/HASecurityUtils.java

```diff
@@ -0,0 +1,291 @@
+        getSCMCertChain(scmNodeDetailsProto, getEncodedString(csr));
+    String pemEncodedCert = response.getX509Certificate();
+
+    try {
+
+      // Store SCM CA certificate.
+      if (response.hasX509CACertificate()) {
+        String pemEncodedRootCert = response.getX509CACertificate();
+        client.storeCertificate(pemEncodedRootCert, true, true);
+        client.storeCertificate(pemEncodedCert, true);
+
+
+        // Write to the location, so that Default CA server can read the
+        // certificate generated by cert client.
+        CertificateCodec certCodec =
+            new CertificateCodec(new SecurityConfig(config),
+                client.getComponentName());
+
+        X509Certificate certificate =
+            CertificateCodec.getX509Certificate(pemEncodedCert);
+        certCodec.writeCertificate(certCodec.getCertificateHolder(certificate));
+
+        // Persist scm cert serial ID.
+        scmStorageConfig.setScmCertSerialId(certificate.getSerialNumber()
+            .toString());
+      } else {
+        throw new RuntimeException("Unable to retrieve SCM certificate chain");
+      }
+    } catch (IOException | CertificateException e) {
+      LOG.error("Error while storing SCM signed certificate.", e);
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void getPrimarySCMSignedCert(CertificateClient client,
+      OzoneConfiguration config, String fetchedSCMId,
+      SCMStorageConfig scmStorageConfig, InetSocketAddress scmAddress)
+      throws IOException {
+
+    PKCS10CertificationRequest csr = generateCSR(client, scmStorageConfig,
+        config, scmAddress, fetchedSCMId);
+
+    CertificateServer rootCAServer =
+        initializeRootCertificateServer(scmStorageConfig.getClusterID(),
+            scmStorageConfig.getScmId(), null);
+
+    rootCAServer.init(new SecurityConfig(config),
+        CertificateServer.CAType.SELF_SIGNED_CA);
+
+    try {
+      X509CertificateHolder certificateHolder = rootCAServer.
+          requestCertificate(csr, KERBEROS_TRUSTED, SCM).get();
+
+      X509CertificateHolder rootCAServerCACertificateHolder =
+          rootCAServer.getCACertificate();
+
+      String pemEncodedCert =
+          CertificateCodec.getPEMEncodedString(certificateHolder);
+
+      String pemEncodedRootCert =
+          CertificateCodec.getPEMEncodedString(rootCAServerCACertificateHolder);
+
+
+      client.storeCertificate(pemEncodedRootCert, true, true);
+      client.storeCertificate(pemEncodedCert, true);
+
+      // Write to the location, so that Default CA server can read the
+      // certificate generated by cert client.
+      CertificateCodec certCodec =
+          new CertificateCodec(new SecurityConfig(config),
+              client.getComponentName());
+
+      certCodec.writeCertificate(certificateHolder);
+
+        // Persist scm cert serial ID.
+      scmStorageConfig.setScmCertSerialId(certificateHolder.getSerialNumber()
```

Comment:
Should we persist this to version file as root CA certID, which is different from normal SCM certid.

Reply:
This is not rootCA CertID, this is the signed cert SCM got from root CA.
Similar to other roles like OM/DN only persisting cert serial ID which it got from CA. (In case of SCM it is from root CA, where as from OM/DN it is from Sub SCM CA)